clients/src/main/java/org/apache/kafka/clients/admin/DescribeConsumerGroupsResult.java

```diff
@@ -34,39 +35,40 @@
 @InterfaceStability.Evolving
 public class DescribeConsumerGroupsResult {
 
-    private final Map<String, KafkaFuture<ConsumerGroupDescription>> futures;
+    private final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures;
 
-    public DescribeConsumerGroupsResult(final Map<String, KafkaFuture<ConsumerGroupDescription>> futures) {
+    public DescribeConsumerGroupsResult(final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures) {
         this.futures = futures;
     }
 
     /**
      * Return a map from group id to futures which yield group descriptions.
      */
     public Map<String, KafkaFuture<ConsumerGroupDescription>> describedGroups() {
-        return futures;
+        Map<String, KafkaFuture<ConsumerGroupDescription>> describedGroups = new HashMap<>();
+        for (Map.Entry<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> entry : futures.entrySet()) {
+            describedGroups.put(entry.getKey().idValue, entry.getValue());
+        }
+        return describedGroups;
     }
 
     /**
      * Return a future which yields all ConsumerGroupDescription objects, if all the describes succeed.
      */
     public KafkaFuture<Map<String, ConsumerGroupDescription>> all() {
         return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).thenApply(
-            new KafkaFuture.BaseFunction<Void, Map<String, ConsumerGroupDescription>>() {
-                @Override
-                public Map<String, ConsumerGroupDescription> apply(Void v) {
-                    try {
-                        Map<String, ConsumerGroupDescription> descriptions = new HashMap<>(futures.size());
-                        for (Map.Entry<String, KafkaFuture<ConsumerGroupDescription>> entry : futures.entrySet()) {
-                            descriptions.put(entry.getKey(), entry.getValue().get());
-                        }
-                        return descriptions;
-                    } catch (InterruptedException | ExecutionException e) {
-                        // This should be unreachable, since the KafkaFuture#allOf already ensured
-                        // that all of the futures completed successfully.
-                        throw new RuntimeException(e);
+            nil -> {
+                Map<String, ConsumerGroupDescription> descriptions = new HashMap<>(futures.size());
+                try {
+                    for (Map.Entry<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> entry : futures.entrySet()) {
```

Comment:
Nit: We could also use the lambda expression instead:

```java
    public KafkaFuture<Map<String, ConsumerGroupDescription>> all() {
        return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).thenApply(
            nil -> {
                Map<String, ConsumerGroupDescription> descriptions = new HashMap<>(futures.size());
                futures.forEach((key, future) -> {
                    try {
                        descriptions.put(key.idValue, future.get());
                    } catch (InterruptedException | ExecutionException e) {
                        // This should be unreachable, since the KafkaFuture#allOf already ensured
                        // that all of the futures completed successfully.
                        throw new RuntimeException(e);
                    }
                });
                return descriptions;
            });
    }
```